### PR TITLE
Search keywords bold

### PR DESCRIPTION
**Pull request by danamansana**
This branch changes bolding of keywords to bold keywords individually

### CODE REVIEW

File: src/app/components/ResultsItem/ResultsItem.jsx [hunk 1]
```diff
@@ -23,8 +23,10 @@ class ResultsItem extends React.Component {
     const {
       searchKeyword,
     } = this.props;
+    const escape = s => s.replace(/[-/\\^$*+?.()|[\]{}]/g, '\\$&');
```
Comment (ktp242) [inline src/app/components/ResultsItem/ResultsItem.jsx]: This regular expression looks intimidating lol
I think it worths to explain what this is for if you want to implement this way.

Also I think the method to bold the keywords can be taken out as an independent function.

File: src/app/components/ResultsItem/ResultsItem.jsx [hunk 2]
```diff
@@ -23,8 +23,10 @@ class ResultsItem extends React.Component {
...
+    const searchKeywords = searchKeyword.split(/\s+/).map(escape).join('|');
```
[the remark this answers is not in the document]
Comment (ktp242) [reply src/app/components/ResultsItem/ResultsItem.jsx]: ohhh is it for doing OR?

File: src/app/components/ResultsItem/ResultsItem.jsx (hunk 1 above)
Comment (danamansana) [reply src/app/components/ResultsItem/ResultsItem.jsx]: Good idea. I can't believe JavaScript doesn't have some built in escaping function but that's how it is

**Comment by ktp242** [comment]
If I search `book for kids`, then all the `for`s in the snippet will become bold, is this correct?

**Comment by danamansana** [comment]
Yes, we are treating all words as search words. We could try to make an exception for small words but this would be a little more complicated and require us to get have an agreed list of small words

**Comment by danamansana** [comment]
Also we are bolding subwords. For example, if you search for 'cat', one of the results is 'cats in the catalog', and the second 'cat' is bold. It would be tricky to avoid bolding the second one while still bolding the first one.

**Pull request closed without merging**